plutarch-test/goldens/plutustype.deconstr.uplc.eval.golden

```diff
@@ -0,0 +1,27 @@
+matching.typed.newtype (program 1.0.0 [#d8799f41abff, #d87a80])
+matching.typed.sumtype(ignore-fields) (program 1.0.0 ())
+matching.typed.sumtype(partial-match) (program 1.0.0 [#40])
+matching.typed.sumtype(exhaustive).minting (program 1.0.0 #01)
+matching.typed.sumtype(exhaustive).spending (program 1.0.0 #02)
+matching.typed.sumtype(exhaustive).rewarding (program 1.0.0 #03)
+matching.typed.sumtype(exhaustive).certifying (program 1.0.0 #04)
+matching.typed.sumtype(exhaustive)(ignore-fields).minting (program 1.0.0 #01)
+matching.typed.sumtype(exhaustive)(ignore-fields).spending (program 1.0.0 #02)
+matching.typed.sumtype(exhaustive)(ignore-fields).rewarding (program 1.0.0 #03)
+matching.typed.sumtype(exhaustive)(ignore-fields).certifying (program 1.0.0 #04)
+matching.raw.newtype (program 1.0.0 [#d8799f41abff, #d87a80])
+matching.raw.sumtype(ignore-fields) (program 1.0.0 ())
+matching.raw.sumtype(partial-match) (program 1.0.0 [#40])
+matching.raw.sumtype(exhaustive).minting (program 1.0.0 #01)
+matching.raw.sumtype(exhaustive).spending (program 1.0.0 #02)
+matching.raw.sumtype(exhaustive).rewarding (program 1.0.0 #03)
+matching.raw.sumtype(exhaustive).certifying (program 1.0.0 #04)
+matching.raw.sumtype(exhaustive)(ignore-fields).minting (program 1.0.0 #01)
+matching.raw.sumtype(exhaustive)(ignore-fields).spending (program 1.0.0 #02)
+matching.raw.sumtype(exhaustive)(ignore-fields).rewarding (program 1.0.0 #03)
+matching.raw.sumtype(exhaustive)(ignore-fields).certifying (program 1.0.0 #04)
+fields.typed.extract-single (program 1.0.0 #d87a9f41abff)
+fields.raw.extract-single (program 1.0.0 #d87a9f41abff)
+combined.typed.toValidatorHash (program 1.0.0 (\i0 -> \i0 -> i2 (unBData (force headList [#41ab]))))
+combined.raw.toValidatorHash (program 1.0.0 (\i0 -> \i0 -> i2 (unBData (force headList (force (force sndPair) ( 1
```

Comment:
that's interesting, why are these different?

Reply:
oh boy, it's because the `pmatchRepr` machinery does an extra `plet` to explicitly evaluate the `SndPair` before applying the `pcon . PJust`. But that sort of means the raw version is cheating by doing less work! That's non obvious and I'm not sure whether the raw version should be intentionally crippled here to match the result of the typed one.